diff --git a/frameworks/base/packages/SettingsLib/src/com/android/settingslib/fuelgauge/BatteryStatus.java b/frameworks/base/packages/SettingsLib/src/com/android/settingslib/fuelgauge/BatteryStatus.java
index b3205d7563b..d1478532f4e 100644
--- a/frameworks/base/packages/SettingsLib/src/com/android/settingslib/fuelgauge/BatteryStatus.java
+++ b/frameworks/base/packages/SettingsLib/src/com/android/settingslib/fuelgauge/BatteryStatus.java
@@ -27,7 +27,7 @@ import static android.os.BatteryManager.EXTRA_MAX_CHARGING_VOLTAGE;
 import static android.os.BatteryManager.EXTRA_PLUGGED;
 import static android.os.BatteryManager.EXTRA_PRESENT;
 import static android.os.BatteryManager.EXTRA_STATUS;
-
+import android.util.Log;
 import android.content.Context;
 import android.content.Intent;
 import android.os.BatteryManager;
@@ -72,7 +72,11 @@ public class BatteryStatus {
 
         final int maxChargingMicroAmp = batteryChangedIntent.getIntExtra(EXTRA_MAX_CHARGING_CURRENT,
                 -1);
+        
+        Log.d("BatteryDebug", "zxff maxChargingMicroAmp (μA): " + maxChargingMicroAmp);
+
         int maxChargingMicroVolt = batteryChangedIntent.getIntExtra(EXTRA_MAX_CHARGING_VOLTAGE, -1);
+        Log.d("BatteryDebug", "zxff maxChargingMicroVolt (μV): " + maxChargingMicroVolt);
 
         if (maxChargingMicroVolt <= 0) {
             maxChargingMicroVolt = DEFAULT_CHARGING_VOLTAGE_MICRO_VOLT;
@@ -82,6 +86,7 @@ public class BatteryStatus {
             // to maintain precision equally on both factors.
             maxChargingWattage = (maxChargingMicroAmp / 1000)
                     * (maxChargingMicroVolt / 1000);
+            Log.d("BatteryDebug", "zxff maxChargingWattage (μw): " + maxChargingWattage);
         } else {
             maxChargingWattage = -1;
         }
