diff --git a/custom/mt6771/t8376_l40mmeiw_T10_64_k419/lcm/kernel/YN1010FH007Q_05/YN1010FH007Q_05.c b/custom/mt6771/t8376_l40mmeiw_T10_64_k419/lcm/kernel/YN1010FH007Q_05/YN1010FH007Q_05.c
index 5b1b185ba44..37fae2af715 100755
--- a/custom/mt6771/t8376_l40mmeiw_T10_64_k419/lcm/kernel/YN1010FH007Q_05/YN1010FH007Q_05.c
+++ b/custom/mt6771/t8376_l40mmeiw_T10_64_k419/lcm/kernel/YN1010FH007Q_05/YN1010FH007Q_05.c
@@ -536,7 +536,7 @@ static void lcm_get_params(struct LCM_PARAMS *params)
 }
 static void lcm_init_lcm(void)
 {
-	pr_notice("[KE/LCM] %s() enter\n", __func__);
+	pr_notice("zxfjiu[KE/LCM] %s() enter\n", __func__);
 	push_table(lcm_init_setting,sizeof(lcm_init_setting) / sizeof(struct LCM_setting_table),1);
 }
 
diff --git a/custom/mt6771/t8376_l40mmeiw_T10_64_k419/lcm/kernel/mt65xx_lcm_list.c b/custom/mt6771/t8376_l40mmeiw_T10_64_k419/lcm/kernel/mt65xx_lcm_list.c
index f990d0f56ab..f3041663a9e 100755
--- a/custom/mt6771/t8376_l40mmeiw_T10_64_k419/lcm/kernel/mt65xx_lcm_list.c
+++ b/custom/mt6771/t8376_l40mmeiw_T10_64_k419/lcm/kernel/mt65xx_lcm_list.c
@@ -52,6 +52,10 @@ struct LCM_DRIVER *lcm_driver_list[] = {
 	&YN1010FH007Q_05_lcm_drv,
 #endif
 
+#if defined(YN1010FH014B)
+	&YN1010FH014B_lcm_drv,
+#endif
+
 /* emdoor add end */
 };
 
diff --git a/custom/mt6771/t8376_l40mmeiw_T10_64_k419/lcm/kernel/mt65xx_lcm_list.h b/custom/mt6771/t8376_l40mmeiw_T10_64_k419/lcm/kernel/mt65xx_lcm_list.h
index b4f9adf317c..9e1b221fca9 100755
--- a/custom/mt6771/t8376_l40mmeiw_T10_64_k419/lcm/kernel/mt65xx_lcm_list.h
+++ b/custom/mt6771/t8376_l40mmeiw_T10_64_k419/lcm/kernel/mt65xx_lcm_list.h
@@ -18,6 +18,7 @@ extern struct LCM_DRIVER WD101HHM40AB_D3_06_lcm_drv;
 extern struct LCM_DRIVER SQ101D_Q5DI404_84H501_lcm_drv;
 extern struct LCM_DRIVER SL101PM32D1605_A03_lcm_drv;
 extern struct LCM_DRIVER YN1010FH007Q_05_lcm_drv;
+extern struct LCM_DRIVER YN1010FH014B_lcm_drv;
 
 #ifdef BUILD_LK
 extern void mdelay(unsigned long msec);
diff --git a/custom/mt6771/t8376_l40mmeiw_T10_64_k419/lcm/lk/YN1010FH007Q_05/YN1010FH007Q_05.c b/custom/mt6771/t8376_l40mmeiw_T10_64_k419/lcm/lk/YN1010FH007Q_05/YN1010FH007Q_05.c
index 85cb03df926..7d1284e071f 100755
--- a/custom/mt6771/t8376_l40mmeiw_T10_64_k419/lcm/lk/YN1010FH007Q_05/YN1010FH007Q_05.c
+++ b/custom/mt6771/t8376_l40mmeiw_T10_64_k419/lcm/lk/YN1010FH007Q_05/YN1010FH007Q_05.c
@@ -317,10 +317,30 @@ static void lcm_init_power(void)
 //	lcm_set_gpio_output(GPIO_LCD_PWM_EN, GPIO_OUT_ONE);
 	MDELAY(10);
 }
-/*
+
 static int lcm_compare_id(void){
-		return 0;
-}*/
+	// unsigned int id1 = 0, id2 = 0, id3 = 0;
+	// unsigned char buffer[3];	
+	// printf("zxf1 jinru \n");
+	// read_reg_v2(0xFC, &buffer[0], 1);
+	// id1 = buffer[0]; 
+
+	// read_reg_v2(0xFD, &buffer[1], 1);
+	// id2 = buffer[1];
+
+	// read_reg_v2(0xFE, &buffer[2], 1);
+	// id3 = buffer[2];
+
+	// printf("zxf1   %s,YN1010FH007Q  id1 =0x%x, id2 = 0x%x, id3 = 0x%x\n", __func__, id1, id2, id3);
+
+    // printf("zxf1 %s, =======emdoor====YN1010FH007Q buffer[0] id1 = %x\n", __func__, buffer[0]);
+    // printf("zxf1 %s, =======emdoor====YN1010FH007Q buffer[1] id2 = %x\n", __func__, buffer[1]);
+    // printf("zxf1 %s, =======emdoor====YN1010FH007Q buffer[2] id3 = %x\n", __func__, buffer[2]);
+    
+
+	printf("zxf1 [LK/LCM] %s\n", __func__);
+		return 1;
+}
 
 LCM_DRIVER YN1010FH007Q_05_lcm_drv = {
 	.name           	= "YN1010FH007Q_05",
@@ -328,6 +348,6 @@ LCM_DRIVER YN1010FH007Q_05_lcm_drv = {
 	.get_params         = lcm_get_params,
 	.init_power			= lcm_init_power,
 	.init               = lcm_init,
-	//.compare_id     = lcm_compare_id,
+	// .compare_id         = lcm_compare_id,
 };
 
diff --git a/custom/mt6771/t8376_l40mmeiw_T10_64_k419/lcm/lk/YN1010FH014B/YN1010FH014B.c b/custom/mt6771/t8376_l40mmeiw_T10_64_k419/lcm/lk/YN1010FH014B/YN1010FH014B.c
index 9bdbac43f64..413d4fdae78 100644
--- a/custom/mt6771/t8376_l40mmeiw_T10_64_k419/lcm/lk/YN1010FH014B/YN1010FH014B.c
+++ b/custom/mt6771/t8376_l40mmeiw_T10_64_k419/lcm/lk/YN1010FH014B/YN1010FH014B.c
@@ -475,7 +475,7 @@ static void lcm_init(void)
 {
 	int ret = 0;
 
-	printf("zxfxin[LK/LCM] %s\n", __func__);
+	printf("[LK/LCM] %s\n", __func__);
 
 	push_table(lcm_init_setting,
 		sizeof(lcm_init_setting) / sizeof(struct LCM_setting_table),
@@ -503,7 +503,26 @@ static void lcm_init_power(void)
 }
 
 static int lcm_compare_id(void){
-	printf("zxfxin[LK/LCM] %s\n", __func__);
+	unsigned int id1 = 0, id2 = 0, id3 = 0;
+	unsigned char buffer[3];	
+	printf("zxf jinru \n");
+	read_reg_v2(0xFC, &buffer[0], 1);
+	id1 = buffer[0]; 
+
+	read_reg_v2(0xFD, &buffer[1], 1);
+	id2 = buffer[1];
+
+	read_reg_v2(0xFE, &buffer[2], 1);
+	id3 = buffer[2];
+
+	printf("zxf   %s,YN1010FH014B  id1 =0x%x, id2 = 0x%x, id3 = 0x%x\n", __func__, id1, id2, id3);
+
+    printf("zxf %s, =======emdoor====YN1010FH014B buffer[0] id1 = %x\n", __func__, buffer[0]);
+    printf("zxf %s, =======emdoor====YN1010FH014B buffer[1] id2 = %x\n", __func__, buffer[1]);
+    printf("zxf %s, =======emdoor====YN1010FH014B buffer[2] id3 = %x\n", __func__, buffer[2]);
+    
+	printf("zxf[LK/LCM] %s\n", __func__);
+	if ()
 	return 1;
 }
 
@@ -513,6 +532,6 @@ LCM_DRIVER YN1010FH014B_lcm_drv = {
 	.get_params         = lcm_get_params,
 	.init_power			= lcm_init_power,
 	.init               = lcm_init,
-	.compare_id         = lcm_compare_id,
+	// .compare_id         = lcm_compare_id,
 };
 
diff --git a/custom/mt6771/t8376_l40mmeiw_T10_64_k419/lcm/lk/mt65xx_lcm_list.c b/custom/mt6771/t8376_l40mmeiw_T10_64_k419/lcm/lk/mt65xx_lcm_list.c
index 1ccf33a9a8c..b937a02397c 100755
--- a/custom/mt6771/t8376_l40mmeiw_T10_64_k419/lcm/lk/mt65xx_lcm_list.c
+++ b/custom/mt6771/t8376_l40mmeiw_T10_64_k419/lcm/lk/mt65xx_lcm_list.c
@@ -55,6 +55,7 @@ extern LCM_DRIVER WD101HHM40AB_D3_06_lcm_drv;
 extern LCM_DRIVER SQ101D_Q5DI404_84H501_lcm_drv;
 extern LCM_DRIVER SL101PM32D1605_A03_lcm_drv;
 extern LCM_DRIVER YN1010FH007Q_05_lcm_drv;
+extern LCM_DRIVER YN1010FH014B_lcm_drv;
 /* emdoor add end */
 extern LCM_DRIVER es6311_anx6585_zigzag_wxga_lcm_drv;
 
@@ -89,6 +90,10 @@ LCM_DRIVER *lcm_driver_list[] = {
 	&YN1010FH007Q_05_lcm_drv,
 #endif
 
+#if defined(YN1010FH014B)
+	&YN1010FH014B_lcm_drv,
+#endif
+
 /* emdoor add end */
 #if defined(ES6311_ANX6585_ZIGZAG_WXGA)
 	&es6311_anx6585_zigzag_wxga_lcm_drv,
diff --git a/custom/mt6771/t8376_l40mmeiw_T10_64_k419/tp_conf/gslX68X/mtk_gslX680.h b/custom/mt6771/t8376_l40mmeiw_T10_64_k419/tp_conf/gslX68X/mtk_gslX680.h
index 981220ab000..d6fa5c48231 100755
--- a/custom/mt6771/t8376_l40mmeiw_T10_64_k419/tp_conf/gslX68X/mtk_gslX680.h
+++ b/custom/mt6771/t8376_l40mmeiw_T10_64_k419/tp_conf/gslX68X/mtk_gslX680.h
@@ -1,3 +1,3 @@
 /*--------------------start----------------------------*/
-#include "/mnt/MT87xx_S0/MT87xx_S0.mp1/kernel-4.19/../custom/mt6771/t8376_l40mmeiw_T10_64_k419/tp_conf/gslX68X/cust_config/MJK_CG101_1562_ISCAN_V1.h"
+#include "/mnt/disk1/MT87xx_S0/MT87xx_S0.mp1/kernel-4.19/../custom/mt6771/t8376_l40mmeiw_T10_64_k419/tp_conf/gslX68X/cust_config/WM1058S.h"
 ///*--------------------end----------------------------*/
diff --git a/kernel-4.19/arch/arm64/configs/t8376_l40mmeiw_T10_64_k419_defconfig b/kernel-4.19/arch/arm64/configs/t8376_l40mmeiw_T10_64_k419_defconfig
index 4bc33d4c652..9d94c8726a4 100755
--- a/kernel-4.19/arch/arm64/configs/t8376_l40mmeiw_T10_64_k419_defconfig
+++ b/kernel-4.19/arch/arm64/configs/t8376_l40mmeiw_T10_64_k419_defconfig
@@ -91,7 +91,7 @@ CONFIG_MTK_PMIC_WRAP_HAL=y
 CONFIG_CUSTOM_KERNEL_IMGSENSOR="gc5035r_mipi_raw gc02m2_mipi_raw"
 CONFIG_MTK_VIDEOCODEC_DRIVER=y
 CONFIG_MTK_LCM=y
-CONFIG_CUSTOM_KERNEL_LCM="YN1010FH007Q_05"
+CONFIG_CUSTOM_KERNEL_LCM="YN1010FH007Q_05 YN1010FH014B"
 CONFIG_MTK_SYNC=y
 CONFIG_MTK_FB=y
 CONFIG_MTK_LCM_PHYSICAL_ROTATION="0"
diff --git a/vendor/mediatek/proprietary/bootable/bootloader/lk/project/t8376_l40mmeiw_T10_64_k419.mk b/vendor/mediatek/proprietary/bootable/bootloader/lk/project/t8376_l40mmeiw_T10_64_k419.mk
index 3777ba741b1..2aecf92dce6 100755
--- a/vendor/mediatek/proprietary/bootable/bootloader/lk/project/t8376_l40mmeiw_T10_64_k419.mk
+++ b/vendor/mediatek/proprietary/bootable/bootloader/lk/project/t8376_l40mmeiw_T10_64_k419.mk
@@ -17,7 +17,7 @@ MTK_SECURITY_SW_SUPPORT = yes
 MTK_MT6370_PMU_CHARGER_SUPPORT := yes
 MTK_MT6370_PMU_BLED_SUPPORT := yes
 MTK_LCM_PHYSICAL_ROTATION = 90
-CUSTOM_LK_LCM="YN1010FH007Q_05"
+CUSTOM_LK_LCM="YN1010FH007Q_05 YN1010FH014B"  
 #nt35595_fhd_dsi_cmd_truly_nt50358 = yes
 MTK_VERIFIED_BOOT_SUPPORT = no
 MTK_SEC_FASTBOOT_UNLOCK_SUPPORT = yes
